### PR TITLE
Cleanup and resynchronize Thread/InternalThread fields.

### PR DESCRIPTION
**Pull request by jaykrell**
- Remove unused.
 - Add missing that just happened to work.
 - Synchronize size between netcore and regular. `#if` considered bad.

Extracted from https://github.com/mono/mono/pull/15859.

### CODE REVIEW

**Comment by jaykrell** [comment]
@monojenkins build deb with monolite

**Comment by jaykrell** [comment]
@monojenkins build failed

File: netcore/System.Private.CoreLib/src/System.Threading/Thread.cs [hunk 1]
```diff
@@ -59,6 +57,7 @@ partial class Thread
 		private IntPtr suspended_event;
 		private int self_suspended;
 		private IntPtr thread_state;
+
 		private Thread self;
 		private object pending_exception;
 		private object start_obj;
```
Comment (lambdageek) [inline netcore/System.Private.CoreLib/src/System.Threading/Thread.cs]: I realize this is not part of the PR, but`pending_exception` and `start_obj` are swapped between netcore managed and native.

Comment (jaykrell) [reply netcore/System.Private.CoreLib/src/System.Threading/Thread.cs]: Good catch. Maybe should really mechanize this soon.
